app/components/pay-page/choose-membership-plan-modal/plan-card.ts

```diff
@@ -0,0 +1,73 @@
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import type TimeService from 'codecrafters-frontend/services/time';
+import { formatTimeDurationForCountdown } from 'codecrafters-frontend/utils/time-formatting';
+import type RegionalDiscountModel from 'codecrafters-frontend/models/regional-discount';
+import type PromotionalDiscountModel from 'codecrafters-frontend/models/promotional-discount';
+import type { PricingPlan } from '../choose-membership-plan-modal';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    activeDiscountForYearlyPlan?: PromotionalDiscountModel | null;
```

Comment:
If we already know what plan we're passing in, we can just pass in `promotionalDiscount` and `regionalDiscount` (just pass in `promotionalDiscount` as empty from the caller). 

We don't need to pass in `activeDiscountForYearlyPlan` and then filter out applicability here since all plan cards other than the yearly plan card don't need access to the active discount for the yearly plan